src/log.cpp

```diff
@@ -229,45 +227,50 @@ const std::string Logger::getLevelLabel(LogLevel lev)
 		"VERBOSE",
 		"TRACE",
 	};
-	assert(lev < LL_MAX && lev >= 0);
 	static_assert(ARRLEN(names) == LL_MAX,
 		"mismatch between loglevel names and enum");
+	assert(lev < LL_MAX && lev >= 0);
 	return names[lev];
 }
 
 LogColor Logger::color_mode = LOG_COLOR_AUTO;
 
-const std::string Logger::getThreadName()
+const std::string &Logger::getThreadName()
 {
-	std::map<std::thread::id, std::string>::const_iterator it;
-
 	std::thread::id id = std::this_thread::get_id();
-	it = m_thread_names.find(id);
+
+	auto it = m_thread_names.find(id);
 	if (it != m_thread_names.end())
 		return it->second;
 
-	std::ostringstream os;
-	os << "#0x" << std::hex << id;
-	return os.str();
+	thread_local std::string fallback_name;
+	if (!fallback_name.empty()) {
```

Comment:
isn't this the wrong way around?

Reply:
Ah, yes. I noticed this before actually but forgot to fix it.